Source/spells.cpp

```diff
@@ -250,7 +250,7 @@ void DoResurrect(int pnum, int rid)
 	int hp;
 
 	if ((char)rid != -1) {
-		AddMissile(plr[rid].position.tile.x, plr[rid].position.tile.y, plr[rid].position.tile.x, plr[rid].position.tile.y, 0, MIS_RESURRECTBEAM, TARGET_MONSTERS, pnum, 0, 0);
+		AddMissile(plr[rid].position.tile, plr[rid].position.tile, (Direction)0, MIS_RESURRECTBEAM, TARGET_MONSTERS, pnum, 0, 0);

```

Comment:
We could do that, but it obfuscates the fact that the direction doesn't matter in this case.

Maybe a new constant, like DIR_DEFAULT is in order?

Reply:
I defaulted to DIR_S as this constant is the only value carrying the same meaning for base Diablo directions and Hellfire directions. Given the type of this function now implies it takes a direction it seemed more appropriate to provide an explicit direction. Especially since `AddMissile()` does pass the value to `SetMissDir()` as if it was a direction even though it often doesn't end up getting used.

That said, like you point out we can work out which AddProc is actually getting called based on the missile type so it would be reasonable to flag it as being unused. I wouldn't want to add anything to the Direction enum in its current form though. Perhaps the outcome of https://github.com/diasurgical/devilutionX/pull/2115#discussion_r650710739 could be applied here and similar call sites?

(Sorry for the lack of context, I went from bottom to top when reviewing and forgot to update this comment.)